go/libraries/doltcore/sqle/enginetest/dolt_queries.go

```diff
@@ -2119,6 +2119,135 @@ var DoltCheckoutScripts = []queries.ScriptTest{
 			},
 		},
 	},
+	{
+		Name: "dolt_checkout and base name resolution",
+		SetUpScript: []string{
+			"create table t (a int primary key, b int);",
+			"call dolt_commit('-Am', 'creating table t');",
+			"call dolt_branch('b2');",
+			"call dolt_branch('b3');",
+			"insert into t values (1, 1);",
+			"call dolt_commit('-Am', 'added values on main');",
+			"call dolt_checkout('b2');",
+			"insert into t values (2, 2);",
+			"call dolt_commit('-am', 'added values on b2');",
+			"call dolt_checkout('b3');",
+			"insert into t values (3, 3);",
+			"call dolt_commit('-am', 'added values on b3');",
+			"call dolt_checkout('main');",
+		},
+		Assertions: []queries.ScriptTestAssertion{
+			{
+				Query:    "select active_branch();",
+				Expected: []sql.Row{{"main"}},
+			},
+			{
+				Query:    "select * from t;",
+				Expected: []sql.Row{{1, 1}},
+			},
+			{
+				Query:            "use `mydb/b2`;",
+				SkipResultsCheck: true,
+			},
+			{
+				Query:    "select active_branch();",
+				Expected: []sql.Row{{"b2"}},
+			},
+			{
+				Query:    "select * from t;",
+				Expected: []sql.Row{{2, 2}},
+			},
+			{
+				Query:    "select * from mydb.t;",
+				Expected: []sql.Row{{1, 1}},
+			},
```

Comment:
huh? `select * from t` is not equivalent to qualifying the table with the database name? That seems really surprising and likely to break basic assumptions SQL users have. I think you gotta have `use` change the checked out branch or it's just gunna come back to us as customer questions/issues.

Reply:
Discussed offline. Agree this could be surprising, but after a lot of rounds of design and experiments around this, I'm convinced this is the only way this can work and be consistent in all contexts. I'll keep thinking about how it could be made easier to understand.

Reply:
Agreed; thanks for discussing. I think my mental model for `use mydb/branch` was thinking of it as using the `mydb` database and checking out the `branch` branch. I know we're actively changing that of course, and if you think of `mydb` as a completely separate/distinct database from `mydb/branch` then this new approach does make a lot of sense. I think we just have to explain that in the right way in docs and see if customers still trip over it.